cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -687,20 +692,26 @@ installExes verbosity baseCtx buildCtx platform compiler
 installLibraries
   :: Verbosity
   -> ProjectBuildContext
+  -> PI.PackageIndex InstalledPackageInfo
   -> Compiler
   -> PackageDBStack
   -> FilePath -- ^ Environment file
   -> [GhcEnvironmentFileEntry]
   -> IO ()
-installLibraries verbosity buildCtx compiler
+installLibraries verbosity buildCtx installedIndex compiler
                  packageDbs envFile envEntries = do
   if supportsPkgEnvFiles $ getImplInfo compiler
     then do
       let
+        getLatest = (=<<) (maybeToList . safeHead . snd) . take 1 . sortBy (comparing (Down . fst))
+                  . PI.lookupPackageName installedIndex
```

Comment:
This might be universally useful, how about adding this function to `PackageIndex` with documentation?

Reply:
I was hoping we could backport this into the cabal 3.10 series, which last I understood, we hoped to release just `cabal-install` without a new `Cabal` library...

Reply:
We'd need to have a fresh look at the planned fixes to make sure it's possible, but not releasing a new Cabal version would indeed be preferred.